### PR TITLE
Update sbt-osgi to 0.10.0

### PR DESCRIPTION
**Pull request by mdedetrich**
This PR brings in the latest version of sbt-osgi. Aside from the package name being changed, of note is that since the initialization of keys was properly fixed in https://github.com/sbt/sbt-osgi/pull/121 we now have to correctly enable the `SbtOsgi` plugin on the projects that actually use sbt-osgi.

The reason why this wasn't needed previously is that keys was being inadvertently initialized without initialising the SbtOsgi plugin as [documented](https://github.com/sbt/sbt-osgi?tab=readme-ov-file#using-sbt-osgi) because the incorrect keys was being put into `projectSettings` and we were referencing them in `OSGi` object.

### CODE REVIEW

**Comment by mdedetrich** [comment]
@Roiocam If you want to have a look as well

File: build.sbt [hunk 1]
```diff
@@ -467,6 +470,7 @@ lazy val testkit = pekkoModule("testkit")
   .settings(AutomaticModuleName.settings("pekko.actor.testkit"))
   .settings(OSGi.testkit)
   .settings(initialCommands += "import org.apache.pekko.testkit._")
+  .enablePlugins(SbtOsgi)
```
Comment (He-Pin) [inline build.sbt:473]: Does the tests need osgi too？

File: build.sbt [hunk 2]
```diff
@@ -427,20 +428,22 @@ lazy val slf4j = pekkoModule("slf4j")
...
   .settings(Protobuf.settings)
-  .enablePlugins(BoilerplatePlugin, Jdk9)
+  .enablePlugins(BoilerplatePlugin, Jdk9, SbtOsgi)
 
 lazy val streamTestkit = pekkoModule("stream-testkit")
   .dependsOn(stream, testkit % "compile->compile;test->test")
   .settings(Dependencies.streamTestkit)
   .settings(AutomaticModuleName.settings("pekko.stream.testkit"))
   .settings(OSGi.streamTestkit)
+  .enablePlugins(SbtOsgi)
```
Comment (He-Pin) [inline build.sbt:446]: So does the kit.

File: build.sbt (hunk 1 above)
Comment (mdedetrich) [reply build.sbt:473]: It's inherited from the parent project. Note that if it's  OT done correctly the sbt build won't even load.

Comment (mdedetrich) [reply build.sbt:473]: Also the tests should not have OSGi enabled anyways, OSGi is about packaging the final jar and we don't package tests

Comment (mdedetrich) [reply build.sbt:473]: Also the tests should not have OSGi enabled anyways, OSGi is about packaging the final jar and we don't package tests

**Comment by He-Pin** [comment]
Before we can merge this, we should validate the Java 9 related classes is right generated

**Comment by mdedetrich** [comment]
> Before we can merge this, we should validate the Java 9 related classes is rightly been generated

Agreed @Roiocam Do you want to do this since you did it last time.

There shouldn't be any issue since this specific issue is pretty black and white (i.e. either sbt loads or it doesn't) but its better to be safe than sorry (again).

**Review by pjfanning — APPROVED** [review]
lgtm

**Review by He-Pin — APPROVED** [review]
lgtm

**Review by jxnu-liguobin — APPROVED** [review]

**Review by Roiocam — APPROVED** [review]
LGTM, Thanks.

I have verified this PR with #1065, it won't break the JDK9 plugin.
```
pekko > jdk9Check
Package check depend on ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,actor-typed) / List(ScopedKey(Scope(Select(ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,actor-typed)), Zero, Zero, Zero),osgiBundle))
Package check depend on ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,cluster-sharding) / List(ScopedKey(Scope(Select(ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,cluster-sharding)), Zero, Zero, Zero),osgiBundle))
Package check depend on ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,remote) / List(ScopedKey(Scope(Select(ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,remote)), Zero, Zero, Zero),osgiBundle))
Package check depend on ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,stream) / List(ScopedKey(Scope(Select(ProjectRef(file:/Users/roiocam/IdeaProject/incubator-pekko/,stream)), Zero, Zero, Zero),osgiBundle))
```

**Pull request merged**